src/btree/bt_sync_obsolete.c

```diff
@@ -406,16 +406,19 @@ __checkpoint_cleanup_page_skip(
      * FIXME: Read internal pages from non-logged tables when the remove/truncate
      * operation is performed using no timestamp.
      */
+    if (addr.type == WT_ADDR_LEAF_NO)
+        *skipp = true;
+    else if (addr.ta.newest_stop_durable_ts == WT_TS_NONE)
+        /* Only process logged tables when checkpoint cleanup is configured to be aggressive. */
+        *skipp = !(F_ISSET(S2C(session), WT_CONN_CKPT_CLEANUP_RECLAIM_SPACE) &&
```

Comment:
This condition is wrong.

Method - None.
1. Logged table - (true & true) - skip.
2. Non-logged table - (true & false) - don't skip (Actually it should skip)

Method - reclaim space
1. Logged table - (false & true) - Do not skip
2. Non-logged table - (true & false) - Do not skip (Actually it should skip)

Reply:
Sorry, the condition is correct. I thought the `!` was only for the flag and not for whole condition.

Reply:
@kommiharibabu, I am happy to change it to:
```
        *skipp = !F_ISSET(S2C(session), WT_CONN_CKPT_CLEANUP_RECLAIM_SPACE) ||
          !F_ISSET(S2BT(session), WT_BTREE_LOGGED);
```
Would that be easier to read?

Reply:
Yes, it may be easier to read the code.

Reply:
Updated in 204385f31b5ff05a56e997fbfedf199a40c046a6